Import tool use wrong mode and generate schema with error

I want to import the following json schema:
```
{
	"$schema": "http://json-schema.org/draft-07/schema#",
	"$id": "https://example.com/schemas/book.json",
	"description": "This is a simple schema to validate a book in a package and you can use template variables with ${{ }}.",
	"type": "object",
	"properties": {
		"title": {
			"type": "string"
		}
	}
}
```
But error occurs if I don't specify mode. 
```
expected 'package', found '{' (and 1 more errors)
```
So I specify the mode to import from jsonschema and I got the following result:
```
"""
This file was generated by the KCL auto-gen tool. DO NOT EDIT.
Editing this file might prove futile when you re-run the KCL auto-gen generate command.
"""
            	            	 
schema Book:
    """
    This is a simple schema to validate a book in a package and you can use template variables with ${{ }}.

    Attributes
    ----------
    title : str, optional
    """

    title?: str
```
I try to run it but get a error:
```
error[E1001]: InvalidSyntax
 --> /tmp/sandbox146949776/prog.k:8:105
  |
8 |     This is a simple schema to validate a book in a package and you can use template variables with ${{ }}.
  |                                                                                                         ^ expected one of ["}"] got eof
  |
```

--- a/pkg/tools/gen/genkcl.go
+++ b/pkg/tools/gen/genkcl.go
@@ -57,8 +57,6 @@ func (k *kclGenerator) GenSchema(w io.Writer, filename string, src interface{}) 
 		codeStr := string(code)
 		var i interface{}
 		switch {
-		case strings.Contains(codeStr, "package "):
-			k.opts.Mode = ModeGoStruct
 		case json.Unmarshal(code, &i) == nil:
 			switch {
 			case strings.Contains(codeStr, "$schema"):
@@ -70,6 +68,8 @@ func (k *kclGenerator) GenSchema(w io.Writer, filename string, src interface{}) 
 			}
 		case yaml.Unmarshal(code, &i) == nil:
 			k.opts.Mode = ModeYaml
+		case strings.Contains(codeStr, "package "):
+			k.opts.Mode = ModeGoStruct
 		default:
 			return errors.New("failed to detect mode")
 		}

--- a/pkg/tools/gen/templates/kcl/schema.gotmpl
+++ b/pkg/tools/gen/templates/kcl/schema.gotmpl
@@ -1,5 +1,5 @@
 schema {{ formatName .Name }}:
-    """{{- "\n" }}
+    r"""{{- "\n" }}
     {{- template "document" . }}
     """{{- "\n" }}
 

--- a/pkg/tools/gen/testdata/jsonschema/additional/expect.k
+++ b/pkg/tools/gen/testdata/jsonschema/additional/expect.k
@@ -4,7 +4,7 @@ Editing this file might prove futile when you re-run the KCL auto-gen generate c
 """
 
 schema Ethernet:
-    """
+    r"""
     Ethernet
 
     Attributes
@@ -17,7 +17,7 @@ schema Ethernet:
     socket?: EthernetSocket = {"HTTP": 80, "HTTPS": 443}
 
 schema EthernetSocket:
-    """
+    r"""
     EthernetSocket
     """
 

--- a/pkg/tools/gen/testdata/jsonschema/basic/expect.k
+++ b/pkg/tools/gen/testdata/jsonschema/basic/expect.k
@@ -4,7 +4,7 @@ Editing this file might prove futile when you re-run the KCL auto-gen generate c
 """
 
 schema Book:
-    """
+    r"""
     Book
 
     Attributes

--- a/pkg/tools/gen/testdata/jsonschema/const/expect.k
+++ b/pkg/tools/gen/testdata/jsonschema/const/expect.k
@@ -4,7 +4,7 @@ Editing this file might prove futile when you re-run the KCL auto-gen generate c
 """
 
 schema Book:
-    """
+    r"""
     Book
 
     Attributes

--- a/pkg/tools/gen/testdata/jsonschema/document/expect.k
+++ b/pkg/tools/gen/testdata/jsonschema/document/expect.k
@@ -4,7 +4,7 @@ Editing this file might prove futile when you re-run the KCL auto-gen generate c
 """
 
 schema MyOrder:
-    """
+    r"""
     Schema for representing an order information.
     It contains the order number, list of items and the current status.
 

--- a/pkg/tools/gen/testdata/jsonschema/items/expect.k
+++ b/pkg/tools/gen/testdata/jsonschema/items/expect.k
@@ -4,7 +4,7 @@ Editing this file might prove futile when you re-run the KCL auto-gen generate c
 """
 
 schema Book:
-    """
+    r"""
     Book
 
     Attributes

--- a/pkg/tools/gen/testdata/jsonschema/multipleof/expect.k
+++ b/pkg/tools/gen/testdata/jsonschema/multipleof/expect.k
@@ -4,7 +4,7 @@ Editing this file might prove futile when you re-run the KCL auto-gen generate c
 """
 
 schema Book:
-    """
+    r"""
     Book
 
     Attributes

--- a/pkg/tools/gen/testdata/jsonschema/nested-items/expect.k
+++ b/pkg/tools/gen/testdata/jsonschema/nested-items/expect.k
@@ -4,7 +4,7 @@ Editing this file might prove futile when you re-run the KCL auto-gen generate c
 """
 
 schema MonacoManifestSchema:
-    """
+    r"""
     MonacoManifestSchema
 
     Attributes
@@ -22,7 +22,7 @@ schema MonacoManifestSchema:
     environment_groups: [MonacoManifestSchemaEnvironmentGroupsItems0]
 
 schema MonacoManifestSchemaEnvironmentGroupsItems0:
-    """
+    r"""
     MonacoManifestSchemaEnvironmentGroupsItems0
 
     Attributes
@@ -37,7 +37,7 @@ schema MonacoManifestSchemaEnvironmentGroupsItems0:
     environments?: [MonacoManifestSchemaEnvironmentGroupsItems0EnvironmentsItems0]
 
 schema MonacoManifestSchemaEnvironmentGroupsItems0EnvironmentsItems0:
-    """
+    r"""
     MonacoManifestSchemaEnvironmentGroupsItems0EnvironmentsItems0
 
     Attributes
@@ -52,7 +52,7 @@ schema MonacoManifestSchemaEnvironmentGroupsItems0EnvironmentsItems0:
     url?: MonacoManifestSchemaEnvironmentGroupsItems0EnvironmentsItems0Url
 
 schema MonacoManifestSchemaEnvironmentGroupsItems0EnvironmentsItems0Url:
-    """
+    r"""
     The URL of this environment
 
     Attributes
@@ -67,7 +67,7 @@ schema MonacoManifestSchemaEnvironmentGroupsItems0EnvironmentsItems0Url:
     value?: str
 
 schema MonacoManifestSchemaProjectsItems0:
-    """
+    r"""
     MonacoManifestSchemaProjectsItems0
 
     Attributes

--- a/pkg/tools/gen/testdata/jsonschema/nested/expect.k
+++ b/pkg/tools/gen/testdata/jsonschema/nested/expect.k
@@ -4,7 +4,7 @@ Editing this file might prove futile when you re-run the KCL auto-gen generate c
 """
 
 schema Book:
-    """
+    r"""
     Book
 
     Attributes
@@ -17,7 +17,7 @@ schema Book:
     author?: BookAuthor
 
 schema BookAuthor:
-    """
+    r"""
     BookAuthor
 
     Attributes

--- a/pkg/tools/gen/testdata/jsonschema/pattern/expect.k
+++ b/pkg/tools/gen/testdata/jsonschema/pattern/expect.k
@@ -5,7 +5,7 @@ Editing this file might prove futile when you re-run the KCL auto-gen generate c
 import regex
 
 schema Cronjob:
-    """
+    r"""
     Cronjob
 
     Attributes

--- a/pkg/tools/gen/testdata/jsonschema/ref/expect.k
+++ b/pkg/tools/gen/testdata/jsonschema/ref/expect.k
@@ -4,7 +4,7 @@ Editing this file might prove futile when you re-run the KCL auto-gen generate c
 """
 
 schema Customer:
-    """
+    r"""
     Customer
 
     Attributes
@@ -17,7 +17,7 @@ schema Customer:
     address?: Address
 
 schema Address:
-    """
+    r"""
     Address
 
     Attributes
@@ -30,7 +30,7 @@ schema Address:
     state?: State
 
 schema State:
-    """
+    r"""
     State
 
     Attributes

--- a/pkg/tools/gen/testdata/jsonschema/unsupport/expect.k
+++ b/pkg/tools/gen/testdata/jsonschema/unsupport/expect.k
@@ -4,7 +4,7 @@ Editing this file might prove futile when you re-run the KCL auto-gen generate c
 """
 
 schema Shop:
-    """
+    r"""
     Schema for representing a shop information.
     In this test case, we use some logic keywords like "oneOf" that can't be directly converted at the moment. To make it still work, we'll convert it into "any" type.
 
@@ -16,14 +16,14 @@ schema Shop:
     products?: any
 
 schema Clothing:
-    """
+    r"""
     Clothing
     """
 
     [...str]: any
 
 schema Product:
-    """
+    r"""
     Product
 
     Attributes

--- a/pkg/tools/gen/testdata/jsonschema/validation/expect.k
+++ b/pkg/tools/gen/testdata/jsonschema/validation/expect.k
@@ -4,7 +4,7 @@ Editing this file might prove futile when you re-run the KCL auto-gen generate c
 """
 
 schema Book:
-    """
+    r"""
     Book
 
     Attributes

--- a/pkg/tools/gen/testdata/terraform/expect.k
+++ b/pkg/tools/gen/testdata/terraform/expect.k
@@ -4,7 +4,7 @@ Editing this file might prove futile when you re-run the KCL auto-gen generate c
 """
 
 schema AlicloudConfigRule:
-    """
+    r"""
     AlicloudConfigRule
 
     Attributes
@@ -17,7 +17,7 @@ schema AlicloudConfigRule:
     resource_types_scope?: [str]
 
 schema AlicloudDbInstance:
-    """
+    r"""
     AlicloudDbInstance
 
     Attributes
@@ -40,7 +40,7 @@ schema AlicloudDbInstance:
         isunique(security_ips)
 
 schema ComplianceItem:
-    """
+    r"""
     ComplianceItem
 
     Attributes

